Source/WebKit/UIProcess/WebProcessCache.cpp

```diff
@@ -189,22 +189,24 @@ bool WebProcessCache::addProcess(Ref<CachedProcess>&& cachedProcess)
 
 void WebProcessCache::evictAtRandomIfNeeded()
 {
-    while (size() >= capacity()) {
-        {
+    while (size() >= capacity() && (m_sharedProcessesPerSite.isEmpty() || m_processesPerSite.isEmpty())) {
```

Comment:
Not sure I understand the logic behind `&& (m_sharedProcessesPerSite.isEmpty() || m_processesPerSite.isEmpty())`. Doesn't that mean that we will never evict if both of the maps are non-empty?

Reply:
Maybe it would be easier to define capacity as an inclusive limit, then you could write:

```
while (size() > capacity()) {
   ...
   if (size() <= capacity()) break;
   ...
}
```

Reply:
Oh oops, I need to put ! in front of each isEmpty check lol.